libkbfs/tlf_journal.go

```diff
@@ -1201,11 +1199,10 @@ func (j *tlfJournal) removeFlushedMDEntry(ctx context.Context,
 }
 
 func (j *tlfJournal) flushOneMDOp(
-	ctx context.Context, end MetadataRevision,
-	maxMDRevToFlush MetadataRevision) (flushed bool, err error) {
+	ctx context.Context, maxMDRevToFlush MetadataRevision) (
+	flushed bool, err error) {
 	if maxMDRevToFlush == MetadataRevisionUninitialized {
-		// Short-cut `getNextMDEntryToFlush`, which would otherwise read
-		// an MD from disk and sign it unnecessarily.
+		// Nothing to do.
```

Comment:
Right, but now the check against `maxMDRevToFlush` happens before signing instead of after. Sorry, I misspoke above, the new code is:

1) Read MD,
2) Check against `maxMDRevToFlush`,
3) Sign MD.

So without the `...Uninitialized` check, we would call `getNextMDEntryToFlush(1)`, which would fetch the earliest MD, but check against 1 and return before signing. So the `...Uninitialized` check just avoids fetching the earliest MD.

Does that make sense?